query/src/main/java/org/infinispan/search/mapper/work/impl/SearchIndexerImpl.java

```diff
@@ -1,39 +1,84 @@
+import org.infinispan.query.backend.QueryInterceptor;
+import org.infinispan.query.logging.Log;
 import org.infinispan.search.mapper.mapping.EntityConverter;
 import org.infinispan.search.mapper.session.impl.InfinispanIndexedTypeContext;
 import org.infinispan.search.mapper.session.impl.InfinispanTypeContextProvider;
 import org.infinispan.search.mapper.work.SearchIndexer;
-import org.infinispan.util.concurrent.BlockingManager;
+import org.infinispan.util.logging.LogFactory;
+
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.internal.functions.Functions;
+import io.reactivex.rxjava3.processors.FlowableProcessor;
+import io.reactivex.rxjava3.processors.UnicastProcessor;
 
 /**
  * @author Fabio Massimo Ercoli
  */
 public class SearchIndexerImpl implements SearchIndexer {
+   private static final Log log = LogFactory.getLog(QueryInterceptor.class, Log.class);
 
    private final PojoIndexer delegate;
    private final EntityConverter entityConverter;
    private final InfinispanTypeContextProvider typeContextProvider;
 
-   private final SimpleScheduledExecutor offloadingExecutor;
+   private final int maxConcurrency;
+
+   private Disposable processorDisposer;
+
+   private FlowableProcessor<Supplier<Flowable<?>>> requestProcessor;
 
    public SearchIndexerImpl(PojoIndexer delegate, EntityConverter entityConverter,
-         InfinispanTypeContextProvider typeContextProvider, BlockingManager blockingManager) {
+                            InfinispanTypeContextProvider typeContextProvider, int maxConcurrency) {
       this.delegate = delegate;
       this.entityConverter = entityConverter;
       this.typeContextProvider = typeContextProvider;
-      this.offloadingExecutor = InfinispanIndexingExecutorProvider.writeExecutor(blockingManager);
+      this.maxConcurrency = maxConcurrency;
+   }
+
+   @Override
+   public void start() {
+      requestProcessor = UnicastProcessor.<Supplier<Flowable<?>>>create().toSerialized();
+      processorDisposer = requestProcessor.flatMap(Supplier::get, maxConcurrency)
+            .subscribe(Functions.emptyConsumer(),
+                  t -> log.fatal("Indexing encountered a non recoverable error", t));
+   }
+
+   private CompletableFuture<?> sendOperation(Supplier<CompletionStage<?>> actualSupplier) {
+      CompletableFuture<Object> future = new CompletableFuture<>();
+
+      Supplier<Flowable<?>> supplier = () -> {
+         try {
+            actualSupplier.get()
+                  .whenComplete((v, t) -> {
+                     if (t != null) {
+                        future.completeExceptionally(t);
+                     } else {
+                        future.complete(v);
+                     }
+                  });
+         } catch (Throwable t) {
+            future.completeExceptionally(t);
+         }
+         return RxJavaInterop.voidCompletionStageToFlowable(future);
```

Comment:
Actually I need to change this so that if there is an exception in Hibernate processing it doesn't stop the processor and is instead ignored, but still propagated to the end user.